### ISSUE TITLE
`VMAlert.spec.notifier.selector.namespaceSelector` seems have no effect

### ISSUE THREAD

**Issue by umezawatakeshi** [VictoriaMetrics/operator#569]
VMAlert is able to use VMAlertmanagers as notifiers instead of specifing notification URLs. The VMAlertmanagers are specified by `VMAlert.spec.notifier.selector` or `VMAlert.spec.notifiers[].selector`, which have type `DiscoverySelector`. However, its `namespaceSelector` seems have no effect and VMAlertmanagers in **all** namespaces are almost always selected. (`labelSelector` seems have a effect as expected)

--

Probably the following line has a problem: https://github.com/VictoriaMetrics/operator/blob/v0.29.2/controllers/factory/vmalert.go#L89

At this line, if a VMAlertmanager's `deletionTimestamp` is not set, i.e. it is not being deleted, it is added to notifier list without being checked its namespace. The line should be:
```
if !item.DeletionTimestamp.IsZero() || (n.Selector.Namespace != nil && !n.Selector.Namespace.IsMatch(&item)) {
```

**Comment by f41gh7** [comment]
Was added at the v0.30.0 release

**Comment by umezawatakeshi** [comment]
The modification made in commit https://github.com/VictoriaMetrics/operator/commit/ac8e33db494dbb0cb897546db33979ae88689473 is just an addition of parentheses and the process has not changed. I confiemd that the behavior is not changed in v0.30.0.

**Comment by f41gh7** [comment]
Thanks must be fixed at v0.30.1 release

**Comment by umezawatakeshi** [comment]
I confirmed that the behavior was fixed in v0.30.1. thank you.
